packages/commonwealth/client/scripts/models/Thread.ts

```diff
@@ -79,128 +100,194 @@ export class Thread implements IUniqueId {
+        return body;
+      }
+    })();
+    this.associatedReactions = [];
+    const tempReactionIds = (reactions ? reactions.map((r) => r.id) : reactionIds) || [];
+    const tempReactionType = (reactions ? reactions.map((r) => r?.type || r?.reaction) : reactionType) || [];
+    const tempAddressesReacted = (reactions ? reactions.map((r) => r?.address || r?.Address?.address) : addressesReacted) || [];
+    if (
+      tempReactionIds.length > 0 && (
+        tempReactionIds.length === tempReactionType.length &&
+        tempReactionType.length === tempAddressesReacted.length
+      )
+    ) {
+      for (let i = 0; i < tempReactionIds.length; i++) {
+        this.associatedReactions.push({
+          id: tempReactionIds[i],
+          type: tempReactionType[i],
+          address: tempAddressesReacted[i],
+        });
+      }
+    }
+    this.chainEntities = (() => {
+      const chainEntitiesProcessed: ChainEntity[] = [];
+      if (chain_entity_meta) {
+        for (const meta of chain_entity_meta) {
+          const full_entity = Array.from(app.chainEntities.store.values())
+            .flat()
+            .filter((e) => e.id === meta.ce_id)[0];
+          if (full_entity) {
+            if (meta.title) full_entity.title = meta.title;
+            chainEntitiesProcessed.push(full_entity);
+          }
+        }
+      }
+
+      return chainEntitiesProcessed
+        ? chainEntitiesProcessed.map((ce) => {
           return {
             id: +ce.id,
             chain,
             type: ce.type,
-            typeId: ce.type_id || ce.typeId,
+            typeId: (ce as any).type_id || ce.typeId,
             completed: ce.completed,
-            author,
+            author: this.author,
           };
         })
-      : [];
-    this.hasPoll = hasPoll;
-    this.lastEdited = lastEdited;
-    this.markedAsSpamAt = markedAsSpamAt;
-    this.archivedAt = archivedAt;
-    this.lockedAt = lockedAt;
-    this.numberOfComments = numberOfComments || 0;
-    this.associatedReactions = [];
-    if (reactionIds) {
-      for (let i = 0; i < reactionIds.length; i++) {
-        this.associatedReactions.push({
-          id: reactionIds[i],
-          type: reactionType[i],
-          address: addressesReacted[i],
+        : [];
+    })()
+    this.versionHistory = (() => {
+      let versionHistoryProcessed;
+      if (version_history) {
+        versionHistoryProcessed = version_history.map((v) => {
+          if (!v) return;
+          let history;
+          try {
+            history = JSON.parse(v);
+            history.author =
+              typeof history.author === 'string'
+                ? JSON.parse(history.author)
+                : typeof history.author === 'object'
+                  ? history.author
+                  : null;
+            history.timestamp = moment(history.timestamp);
+          } catch (e) {
+            console.log(e);
+          }
+          return history;
```

Comment:
Would anonymous function setters for fields be good?

ex:
```
this.field = (() => {})()
```

Reply:
Ideally we factor out _transformations_ into named helper functions, because we may need to reuse the logic elsewhere. For the time being, those functions can go in or around this file, though.

Reply:
for me IIFE is not a solution. It still is anonymous (does not describe what it does) and it is inline (readability is screwed). As Jake said, these utils might be declared in the same file at the top in this case. Important is that the body of model class won't be polluted

Reply:
I am more inclined towards not over-abstracting in general because it would create the same problem it tries to solve, I mean we have many files that are not always easy to navigate 🤔.  

> Ideally we factor out transformations into named helper functions, because we may need to reuse the logic elsewhere. 

I agree but only if "we already need to" reuse logic - not on the condition "we may need to"

For the sake of closing this, I implemented the change

Reply:
Done